src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -307,25 +303,52 @@ protected function processMigration(Document $project, Document $migration, Log
                     $errorMessages[] = "Error occurred while pushing '{$error->getResourceType()}:{$error->getResourceId()}' to destination with message: '{$error->getMessage()}'";
                 }
 
-                $migrationDocument->setAttribute('errors', $errorMessages);
+                $group->setAttribute('errors', $errorMessages);
+                $this->updateMigrationDocument($group, $projectDocument);
+                $this->updateMigrationDocument($migration, $projectDocument);
+                $log->addBreadcrumb(new Breadcrumb("debug", "migration", "Migration hit stage 'finished' and failed", \microtime(true)));
                 $log->addExtra('migrationErrors', json_encode($errorMessages));
-                $this->updateMigrationDocument($migrationDocument, $projectDocument);
 
                 return;
             }
 
-            $migrationDocument->setAttribute('status', 'completed');
-            $migrationDocument->setAttribute('stage', 'finished');
+            $group->setAttribute('status', 'completed');
             $log->addBreadcrumb(new Breadcrumb("debug", "migration", "Migration hit stage 'finished' and succeeded", \microtime(true)));
+            $this->updateMigrationDocument($group, $project);
+
+            // Check if all other groups are finished, if so set parent document to completed aswell.
+            $groupDocuments = $this->dbForProject->find('migrationsGroup', [Query::equal('migrationId', [$migration->getId()])]);
+
+            $result = 'completed';
+            foreach ($groupDocuments as $document) {
+                if ($document->getId() == $group->getId()) {
```

Comment:
You can disclude this value using a notEqual query ?

[
Query::equal('migrationId', [$migration->getId()])],
Query::notEqual('$id', $group->getId())]
)